esphome/components/i2c/i2c_bus_esp_idf.cpp

```diff
@@ -20,20 +21,71 @@ static const char *const TAG = "i2c.idf";
 void IDFI2CBus::setup() {
   ESP_LOGCONFIG(TAG, "Running setup");
   static i2c_port_t next_port = I2C_NUM_0;
-  port_ = next_port;
-#if SOC_HP_I2C_NUM > 1
-  next_port = (next_port == I2C_NUM_0) ? I2C_NUM_1 : I2C_NUM_MAX;
+  this->port_ = next_port;
+  if (this->port_ == I2C_NUM_MAX) {
+    ESP_LOGE(TAG, "No more than %u buses supported", I2C_NUM_MAX);
+    this->mark_failed();
+    return;
+  }
+
+  if (this->timeout_ > 13000) {
+    ESP_LOGW(TAG, "i2c timeout of %" PRIu32 "us greater than max of 13ms on esp-idf, setting to max", this->timeout_);
```

Comment:
Can be a bit shorter & save a few bytes. 🙂 (`TAG` already tells us it's the IDF I2C code 😉)
```suggestion
    ESP_LOGW(TAG, "Using max allowed timeout: 13 ms");
```
Is this milliseconds or microseconds? I see both `us` and `ms` in the original message. I'm assuming microseconds is correct because 13000.

...also...what a seemingly arbitrary value. 😅